quint/src/cliCommands.ts

```diff
@@ -182,7 +183,13 @@ export function load(args: any): CLIProcedure<LoadedStage> {
 export function parse(loaded: LoadedStage): CLIProcedure<ParsedStage> {
   const { args, sourceCode, path } = loaded
   const parsing = { ...loaded, stage: 'parsing' as stage }
-  return parsePhase1(newIdGenerator(), sourceCode, path)
+  const idgen = newIdGenerator()
+  return parsePhase1(idgen, sourceCode, path)
+    .chain(phase1Data => {
+      const resolver = fileSourceResolver()
```

Comment:
Could we instantiate this resolver inside `parsePhase1b`? It would be great if the top-level calls are as simple as possible since we will have to replicate this in the language server.

Reply:
I agree with the intention. However, the resolver is a very important part here. I can imagine that we would like to load some source files differently very soon, e.g., the standard library. Most likely, this would require a different resolver. So it would be good to propagate the resolver that is constructed elsewhere.